app/src/main/java/com/nutomic/syncthingandroid/activities/FirstStartActivity.java

```diff
@@ -33,28 +39,46 @@
 import com.nutomic.syncthingandroid.R;
 import com.nutomic.syncthingandroid.SyncthingApp;
 import com.nutomic.syncthingandroid.service.Constants;
+import com.nutomic.syncthingandroid.util.PermissionUtil;
 
 import javax.inject.Inject;
 
 public class FirstStartActivity extends Activity {
 
-    private static String TAG = "FirstStartActivity";
+    private enum Slide {
+        INTRO,
+        STORAGE,
+        LOCATION,
+        API_LEVEL_30,
+    };
+
+    private static final Slide[] slideOrder = {
+            Slide.INTRO,
+            Slide.STORAGE,
+            Slide.LOCATION,
+            Slide.API_LEVEL_30,
+    };
+    private static int[] mLayouts = new int[]{
+            R.layout.activity_firststart_slide_intro,
+            R.layout.activity_firststart_slide_storage,
+            R.layout.activity_firststart_slide_location,
+            R.layout.activity_firststart_slide_api_level_30,
+    };
```

Comment:
This is a bit grim.

You could rely on the enum definition order for the ordering, and then do the following for the layout ids:

https://www.baeldung.com/java-enum-values